### DISCUSSION TITLE
How to write a close application script

### DISCUSSION THREAD

**Discussion by nadeemkhedr** [johnlindquist/kit#625] [Q&A]
Want to have a global close application script, when you execute it will close the current application (Close all tabs on browsers, and close window on other applications)

Want the same behavior as clicking the red `x` on macos 

In [yabai](https://github.com/koekeishiya/yabai) you could do that with the following script 
```
yabai -m window --close
```

----

Was following this thread in [yabai](https://github.com/koekeishiya/yabai/issues/84) 

Someone posted this applescript that can close the current application 

Script in scriptkit:
```
await applescript(`
  tell application "System Events" to perform action "AXPress" of (first button whose subrole is "AXCloseButton") of (first window whose subrole is "AXStandardWindow") of (first process whose frontmost is true)
`)
```

This works, it closes the selected window, but doesn't work for everything

For alacritty for some reason it doesn't work

**Comment by johnlindquist** [comment]
I'd have to do some research. Is yabai open-source? Can you call yabai from Script Kit? (I'm in my phone)

I imagine we'll have to write a util in Swift/Rust then exec the util. (It's actually on the road map to build a lot of Rust utils to bundle with Script Kit, but that won't happen until next year)

**Reply by nadeemkhedr** [reply]
Yes its open source and this is the [commit](https://github.com/koekeishiya/yabai/commit/9d2f97f9017b799f5782c99806858df517a62fa0) that added close functionality, it's written in objective-c

**Reply by johnlindquist** [reply]
Thanks, I'll look into it later this week. I haven't used yabai, but it looks like a great tool
